fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java

```diff
@@ -497,6 +501,31 @@ public void execute() throws Exception {
         }
     }
 
+    private void handleCreateTableAsSelectStmt(long beginTimeInNanoSecond) throws Exception {
+        CreateTableAsSelectStmt createTableAsSelectStmt = (CreateTableAsSelectStmt) parsedStmt;
+
+        // if create table failed should not drop table. because table may already exists,
+        // and for other cases the exception will throw and the rest of the code will not be executed.
+        createTableAsSelectStmt.createTable(context);
+        try {
+            InsertStmt insertStmt = createTableAsSelectStmt.getInsertStmt();
+            ExecPlan execPlan = new StatementPlanner().plan(insertStmt, context);
+            handleInsertStmtWithNewPlanner(execPlan, ((CreateTableAsSelectStmt) parsedStmt).getInsertStmt());
+            if (context.getSessionVariable().isReportSucc()) {
```

Comment:
Are query profile and insert profile all gathered?

Reply:
I check the code & test. the profile generate is no problem.
It will generate a profile equivalent to insert into select.